File: server/src/main/java/com/gitittogether/skillForge/server/dto/response/skill/SkillResponse.java
package com.gitittogether.skillForge.server.dto.response.skill;


import com.gitittogether.skillForge.server.dto.response.course.CategoryResponse;
import com.gitittogether.skillForge.server.model.course.Level;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SkillResponse {
    private String id;
    private String name;
    private String description;
    private CategoryResponse category;
    private Level difficultyLevel;
}

File: client/src/types/response/skill/SkillResponse.ts
import { CategoryResponse } from '@/types/response/course/CategoryResponse.ts';
import { Level } from '@/types/utils/Level.ts';

export interface SkillResponse {
  id: string;
  name: string;
  description: string;
  category: CategoryResponse;
  difficultyLevel: Level;
}